### DISCUSSION TITLE
How to access `errors` in `useFieldArray`

### DISCUSSION THREAD

**Discussion by robbypambudi** [react-hook-form/react-hook-form#9698] [Q&A]
This is my current code 

```
 const { fields, append } = useFieldArray({
    name: 'peserta',
    shouldUnregister: true,
    rules: { required: true },
  });
```
```
<Input
    readOnly={true}
    label='Asal Sekolah/Institusi'
    name='asal_sekolah'
    id={`peserta.${index}.asal_sekolah`}
    placeholder='Asal Sekolah'
    helperText='Contoh : SMA Negeri 1 Surabaya'
  />
<Input
  label='Nama Lengkap'
  name='nama'
  id={`peserta.${index}.nama`}
  placeholder='Nama Pendaftar'
/>

<Input
  label='NIK'
  name='nik'
  id={`peserta.${index}.nik`}
  placeholder='Masukkan NIK'
/>
```

and this is my Input Component
![Screen Shot 2023-01-03 at 02 02 05](https://user-images.githubusercontent.com/34505233/210270219-cb624bf1-e357-40d0-8881-94d3af8eba3a.png)

https://user-images.githubusercontent.com/34505233/210270374-056c2aa2-9db0-4754-a1dd-11df3ccfefc8.mov

I don't get an error status and an error message.
anybody can help me?

**Comment by Moshyfawn** [comment]
If want the root field array `required` error, here's how to get it
```js
errors.peserta?.root?.message
```

**Reply by Moshyfawn** [reply]
`useFieldArray` `rules` are applied to the whole field array, not each field: `required` checks that the field array is there, `minLength` / `maxLength` - check the min and max field array length
